tasks/webfont.js

```diff
@@ -389,38 +392,133 @@ module.exports = function(grunt) {
-			// If some styles was not included in CSS we should include them in HTML to properly render icons
+			if (key === 'html') {
+				return prepareHTMLContext(context);
+			}
+
+			logger.warn('Key specified to function `prepareTemplateContext` does not correspond to a particular method.');
+			return context;
+		}
+
+		/*
+		 * Makes custom extends necessary for use with preparing the template context
+		 * object for the HTML demo.
+		 */
+		function prepareHTMLContext(context) {
+
+			var htmlStyles;
+
+			// Prepare relative font paths for injection into @font-face refs in HTML
 			var relativeRe = new RegExp(_s.escapeRegExp(o.relativeFontPath), 'g');
 			var htmlRelativeFontPath = normalizePath(path.relative(o.destHtml, o.dest));
-			var context = _.extend(o, {
-				fontSrc1: o.fontSrc1.replace(relativeRe, htmlRelativeFontPath),
-				fontSrc2: o.fontSrc2.replace(relativeRe, htmlRelativeFontPath),
+			var _fontSrc1 = o.fontSrc1.replace(relativeRe, htmlRelativeFontPath);
+			var _fontSrc2 = o.fontSrc2.replace(relativeRe, htmlRelativeFontPath);
+
+			_.extend(context, {
+				fontSrc1: _fontSrc1,
+				fontSrc2: _fontSrc2,
 				fontfaceStyles: true,
 				baseStyles: true,
 				extraStyles: false,
 				iconsStyles: true,
 				stylesheet: 'css'
 			});
-			var htmlStyles = renderTemplate(o.cssTemplate, context);
-			var htmlContext = _.extend(context, {
+
+			// Prepares CSS for injection into <style> tag at to of HTML
+			htmlStyles = renderTemplate(o.cssTemplate, context);
+			_.extend(context, {
 				styles: htmlStyles
 			});
 
+			return context;
+		}
+
+		/*
+		 * Iterator function used as callback by looping construct below to
+		 * render "custom output" via mini configuration objects specified in
+		 * the array `options.customOutputs`.
+		 */
+		function generateCustomOutput(outputConfig) {
+
+			// Accesses context
+			var context = prepareTemplateContext();
+			_.extend(context, outputConfig.context);
+
+			// Prepares config attributes related to template filepath
+			var templatePath = outputConfig.template;
+			var extension = path.extname(templatePath);
+			var syntax = outputConfig.syntax || '';
+
+			// Renders template with given context
+			var template = readTemplate(templatePath, syntax, extension);
+			var output = renderTemplate(template, context);
+
+			// Prepares config attributes related to destination filepath
+			var dest = outputConfig.dest || o.dest;
+
+			var filepath;
+			var destParent;
+			var destName;
+
+			if (path.extname(dest) === '') {
+				// If user specifies a directory, filename should be same as template
+				destParent = dest;
+				destName = path.basename(outputConfig.template);
+				filepath = path.join(dest, destName);
+			}
+			else {
+				// If user specifies a file, that is our filepath
+				destParent = path.dirname(dest);
+				filepath = dest;
+			}
+
+			// Ensure existence of parent directory and output to file as desired
+			fse.ensureDirSync(destParent);
```

Comment:
We already have mkdirp for that — https://github.com/sapegin/grunt-webfont/blob/master/tasks/webfont.js#L228